app/controllers/wikis_controller.rb

```diff
@@ -0,0 +1,61 @@
+class WikisController < ApplicationController
+    def index
+        @wikis = Wiki.all
+    end
+
+    def show
+        @wiki = Wiki.find(params[:id])
+    end
+
+    def new
+        @wiki = Wiki.new
+    end
+
+    def create
+        # #9
+        @wiki = Wiki.new
+        @wiki.title = params[:wiki][:title]
+        @wiki.body = params[:wiki][:body]
+
+        # #10
+        if @wiki.save
+            # #11
+            flash[:notice] = 'Wiki was saved.'
+            redirect_to @wiki
+        else
+            # #12
+            flash.now[:alert] = 'There was an error saving the wiki. Please try again.'
+            render :new
+        end
+    end
+
+    def edit
+        @wiki = Wiki.find(params[:id])
+    end
+
+    def update
+        @wiki = Wiki.find(params[:id])
+        @wiki.title = params[:wiki][:title]
+        @wiki.body = params[:wiki][:body]
```

Comment:
I'm going to recommend you refactor these two lines into a `wiki_params` methods that uses the params.require(:wiki).permit() method like they did in the checkpoints. this is an important security feature called Strong Parameters, you can read up on if you want. And I can explain it more if you'd like. 

You can still merge this PR, but make the change in your latest branch.